hpaction/build_hpactionvars.py

```diff
@@ -64,10 +66,103 @@ def create_complaint(area: str, description: str) -> hp.TenantComplaints:
+    else:
+        v.landlord_entity_or_individual_mc = hp.LandlordEntityOrIndividualMC.INDIVIDUAL
+
+        # I don't think these are actually used in rendering the form, but just in case...
+        v.landlord_name_first_te = contact.first_name
+        v.landlord_name_last_te = contact.last_name
+    v.landlord_entity_name_te = contact.name
+    v.served_person_te = contact.name
+
+
+def fill_landlord_management_info_from_company(
+    v: hp.HPActionVariables,
+    mgmtco: nycdb.models.Company
+) -> None:
+    v.management_company_address_city_te = mgmtco.address.city
+    v.management_company_address_street_te = mgmtco.address.street_name
+    v.management_company_address_zip_te = mgmtco.address.zipcode
+    v.management_company_address_state_mc = nycdb_addr_to_hp_state(mgmtco.address)
+    v.service_address_full_management_company_te = ", ".join(
+        mgmtco.address.lines_for_mailing)
+    v.served_person_management_company_te = mgmtco.name
+    v.management_company_name_te = mgmtco.name
+
+    # TODO: We might actually want to fill this out even if we don't find
+    # a management company, as this could at least generate the required
+    # forms. Need to find this out.
+    v.management_company_to_be_sued_tf = True
+    v.mgmt_co_is_party_tf = True
+    v.service_already_completed_mgmt_co_tf = False
+    v.hpd_service_mgmt_co_mc = hp.HPDServiceMgmtCoMC.MAIL
+    v.service_method_mgmt_co_mc = hp.ServiceMethodMgmtCoMC.MAIL
+
+
+def fill_landlord_info_from_bbl(v: hp.HPActionVariables, pad_bbl: str) -> bool:
+    landlord_found = False
+    contact = nycdb.models.get_landlord(pad_bbl)
+    if contact:
+        landlord_found = True
+        fill_landlord_info_from_contact(v, contact)
+    mgmtco = nycdb.models.get_management_company(pad_bbl)
+    if mgmtco:
+        fill_landlord_management_info_from_company(v, mgmtco)
+    return landlord_found
+
+
+def fill_landlord_info(v: hp.HPActionVariables, user: JustfixUser) -> None:
+    landlord_found = False
+
+    v.service_already_completed_landlord_tf = False
+    v.hpd_service_landlord_mc = hp.HPDServiceLandlordMC.MAIL
+    v.service_method_mc = hp.ServiceMethodMC.MAIL
+    v.landlord_is_party_tf = True
+
+    if hasattr(user, 'onboarding_info'):
+        pad_bbl: str = user.onboarding_info.pad_bbl
+        if pad_bbl:
+            landlord_found = fill_landlord_info_from_bbl(v, pad_bbl)
+
+    if not landlord_found and hasattr(user, 'landlord_details'):
+        ld = user.landlord_details
+        v.landlord_entity_name_te = ld.name
+        v.served_person_te = ld.name
+        v.service_address_full_te = ld.address
+
+
 def user_to_hpactionvars(user: JustfixUser) -> hp.HPActionVariables:
     v = hp.HPActionVariables()
 
+    # TODO: The HP Action form actually has a field for home phone
+    # and a separate one for work; it's unclear which one the
+    # user may have provided, but we'll assume it's home for now.
```

Comment:
@romeboards also curious what you think of this. Since you filled out "home phone" on LHI's site, I'm guessing it's the one we want to fill out for the user.